File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.js
function solution(arr) {
    const [row, col] = [arr.length, arr[0].length];
    if(row === col) return arr;
    else if(row < col) return [...arr, ...new Array(col - row).fill().map(() => new Array(col).fill(0))];
    else return arr.map((r) => [...r, ...new Array(row - col).fill(0)]);
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    answer = []
    row, col = len(arr), len(arr[0])  # 리스트([]) 대신 튜플로 변경

    if row == col:
        return arr
    elif row < col:
        answer.extend(arr) 
        answer.extend([[0] * col for _ in range(col - row)])
    else:
        for i in range(row):
            arr[i].extend([0] * (row - len(arr[i])))
            answer.append(arr[i])  
    return answer
